fit.py

```diff
@@ -0,0 +1,296 @@
+        recipes, pagination, request.args.get("page")
+    )
+    return render_template(
+        "index.html",
+        page=page,
+        recipes=recipes,
+        next=next,
+        categories=get_all_categories_from_db(),
+        selected_category=selected_category,
+        should_show_background_image=True,
+    )
+
+
+# =================================
+# CREATEUSER MODAL- function will allow create new account
+# ==================================
+@app.route("/create_user", methods=["POST"])
+def createuser():
+    regex = "^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$"  # noqa: W605
+    if not re.search(regex, request.form.get("email")):
+        flash("invalid email")
+        return redirect(url_for("index"))
+
+    newuser = {
+        "username": request.form.get("username"),
+        "password": request.form.get("password"),
+        "email": request.form.get("email"),
+    }
+    user = mydb.users.find_one({"username": request.form["username"]})
+    if user:
+        flash("Username already exists")
+    else:
+
+        try:
+            users_col.insert_one(newuser)
+            flash(
+                "Congratulation "
+                + request.form.get("username")
+                + "! You have created account"
+            )
+        # try and except to notify user in case there was some troubleshooting
+        except Exception:
+            flash(
+                "Sorry, there was some problem, user "
+                + request.form.get("username")
+                + " was not added to database"
+            )
+
+    return redirect(url_for("index"))
+
+
+# =================================
+# LOGIN MODAL- function will allow login to created account
+# ==================================
+@app.route("/login", methods=["POST", "GET"])
+def login():
+    login_user = mydb.users.find_one({"username": request.form["username"]})
+    if login_user:
+        if request.form["password"] == login_user["password"]:
+            session["username"] = request.form["username"]
+            session["logged_in"] = True
+            flash("Welcome " + session["username"])
+            return redirect(url_for("index"))
+        else:
+            flash("Please check your credentials")
+    else:
+        flash("Please create account to login")
+    return redirect(url_for("index"))
+
+
+# =================================
+# LOGOUT MODAL- function will allow logout from account
+# ==================================
+@app.route("/logout", methods=["GET"])
+def logout():
+    if session.get("logged_in"):
+        session["logged_in"] = False
+        session.pop("username")
+        flash("You are logged out!")
+        return redirect(url_for("index"))
+    else:
+        return render_template("index.html")
+
+
+# =============================
+# DISPLAY 'MY RECIPES' SCREEN - Allowing user in input a or view, edit
+# delete only users recipes
+# =============================
+@app.route("/myrecipies")
```

Comment:
Let's use `recipe/<username>` instead. Pass in the username from the template and use it to retrieve the recipe here